### DISCUSSION TITLE
How do I translate the reset password email text?

### DISCUSSION THREAD

**Discussion by seabasss** [Laravel-Lang/common#32] [Q&A] [answered]
I'm trying out the Swedish language package and the following part doesn't get translated properly. Any hints on what I can change to fix it?

```
@lang(
    "If you're having trouble clicking the \":actionText\" button, copy and paste the URL below\n".
    'into your web browser:',
    [
        'actionText' => $actionText,
    ]
) <span class="break-all">[{{ $displayableActionUrl }}]({{ $actionUrl }})</span>
```

is listed in the Swedish translation file as this:

```
"If you’re having trouble clicking the \":actionText\" button, copy and paste the URL below\ninto your web browser:": "Om det ej fungerar att klicka på \":actionText\"-knappen, kopiera och klista in länken nedan i webbläsarens adressfält:\n[:displayableActionUrl](:actionURL)",
```

which in the email shows up as this:

```
If you're having trouble clicking the "Återställ lösenordet" button, copy and paste the URL below into your web browser:http...
```

Thanks!

**Comment by luisprmat** [comment, marked as answer]
[Line 273](https://github.com/Laravel-Lang/lang/blob/master/locales/sv/sv.json#L273) of `locales/sv/sv.json` need to be translated. You can submit a **Pull Request**.

*Possible translation*: `Om det ej fungerar att klicka på \":actionText\"-knappen, kopiera och klista in länken nedan i webbläsarens adressfält` ?

**Comment by seabasss** [comment]
That's what I have in my `sv.json` file already and it's not working.

As you see the line below the one you highlighted already contains the translation, but even with the first line removed it doesn't work.

Ideas?

**Reply by luisprmat** [reply]
The problem is that laravel changed the apostrophe from `’`  to `'` and we add the new line to the translations but the previous one was not deleted to avoid compatibility problems.
You can search the line `If you’re having ...` and copy the translation from `If you're having ...`

**Reply by seabasss** [reply]
Oh, how sneaky! I didn't notice, I thought the lines were identical. Thank you, that did it!

**Reply by luisprmat** [reply]
Or in your translation change the apostrophe

**Comment by seabasss** [comment]
One more thing. This still shows up in the final translated string on the page `:displayableActionUrl`

Using:
```
"If you're having trouble clicking the \":actionText\" button, copy and paste the URL below\ninto your web browser:": "Om det ej fungerar att klicka på \":actionText\"-knappen, kopiera och klista in länken nedan i webbläsarens adressfält:\n[:displayableActionUrl](:actionURL)",
```

Ideas? Thanks!

**Reply by luisprmat** [reply]
Is that part necessary?

**Reply by seabasss** [reply]
Not sure, just going for what the original was:
```
@lang(
    "If you're having trouble clicking the \":actionText\" button, copy and paste the URL below\n".
    'into your web browser:',
    [
        'actionText' => $actionText,
    ]
) <span class="break-all">[{{ $displayableActionUrl }}]({{ $actionUrl }})</span>
```

**Reply by seabasss** [reply]
This is what I ended up with and it works for me :) Thanks!

```
"If you're having trouble clicking the \":actionText\" button, copy and paste the URL below\ninto your web browser:": "Om det ej fungerar att klicka på \":actionText\"-knappen, kopiera och klista in länken nedan i webbläsarens adressfält:\n",
```

**Reply by luisprmat** [reply]
Could you contribute to the community by submitting a PR to this package by putting the new translation on line 273 and deleting the part of `\n[:displayableActionUrl](:actionURL)` that is not needed?

**Reply by seabasss** [reply]
Sure, was gonna wait until I have more stuff translated, but in the meantime, is there a super simple guide for how to do it?

**Reply by luisprmat** [reply]
For now I formulated [this PR](https://github.com/Laravel-Lang/lang/pull/1884) one in particular. You can see the Swedish missing translations in [`docs/statuses/sv.md`](https://github.com/Laravel-Lang/lang/blob/master/docs/statuses/sv.md) and a short guide in [`docs/contributing-to-dev.md#how-can-i-fix-a-file-`](https://github.com/Laravel-Lang/lang/blob/master/docs/contributing-to-dev.md#how-can-i-fix-a-file-)

**Reply by seabasss** [reply]
Thanks!
